### PR TITLE
Feature/52 dependant libraries fixed versions

### PR DESCRIPTION
**Pull request by Ulrond**
#52 corrected patch based on latest checkout
- fixed version of libraries now pulled.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -163,3 +170,9 @@ cleanall: clean
 cleanhost-tools:
 	@$(ECHOE) ${GREEN}Performing Clean on host-tools [$(TOP_DIR)/host-tools]${NC}
 	@${RM} -rf $(TOP_DIR)/host-tools
+
+printenv:
+	@echo "Environment variables: [UT]"
```
Comment (kanjoe24) [inline Makefile:175]: can we add @${ECHOE} for this?

File: configure.sh [hunk 2]
```diff
@@ -35,7 +35,12 @@ pushd ${MY_DIR} > /dev/null
 
 FRAMEWORK_DIR=${MY_DIR}/framework/${TARGET}
 LIBYAML_DIR=${FRAMEWORK_DIR}/libfyaml-master
+LIBYAML_VERSION=997b480cc4239a7f55771535dff52ad69bd4eb5b #30th September 2024
+#LIBYAML_VERSION=592ccc17552ba3eb51b479432986d8786c4fbbe0 #July 23, 2024
```
Comment (kanjoe24) [inline configure.sh:39]: do we need to retain this SHA?

File: Makefile [hunk 3]
```diff
@@ -127,7 +130,7 @@ all: framework
 lib : ${OBJS}
 	@$(ECHOE) ${GREEN}Generating lib [${YELLOW}$(LIB_DIR)/$(TARGET_LIB)${GREEN}]${NC}
 	@$(MKDIR_P) $(LIB_DIR)
-	@$(CC) $(CFLAGS) -o $(LIB_DIR)/$(TARGET_LIB) $^ $(XLDFLAGS)
+	@$(CC) $(XCFLAGS) -o $(LIB_DIR)/$(TARGET_LIB) $^ $(XLDFLAGS)
 
 # Make any c source
 $(BUILD_DIR)/%.o: %.c
```
Comment (kanjoe24) [inline Makefile:136]: Should we also change line: 151, for XCFLAGS?

@$(ECHOE) ${YELLOW}CFLAGS:${NC} ${CFLAGS}

Comment (Ulrond) [reply Makefile:136]: we should still be printing CFLAGS so we know what it is.

File: Makefile (hunk 1 above)
Comment (Ulrond) [reply Makefile:175]: updated

**Review by kanjoe24 — DISMISSED** [review]
Looks good.

**Comment by kanjoe24** [comment]
Results form Test Script:

```
==========================================================
RESULTS for VM-SYNC 
On the branch feature/52-dependant-libraries-fixed-versions. PASS
build/linux/curl/lib/libcurl.a exists. PASS
build/linux/openssl/lib/libssl.a exists. PASS 
host-tools/CMake-3.30.0/build/bin/cmake exists. PASS 
==========================================================
==========================================================
RESULTS for ubuntu 
On the branch feature/52-dependant-libraries-fixed-versions. PASS
build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for VM-SYNC 
On the branch feature/52-dependant-libraries-fixed-versions. PASS
build/linux/curl/lib/libcurl.a exists. PASS
build/linux/openssl/lib/libssl.a exists. PASS 
host-tools/CMake-3.30.0/build/bin/cmake exists. PASS 
==========================================================
==========================================================
RESULTS for dunfell_arm 
On the branch feature/52-dependant-libraries-fixed-versions. PASS
build/arm/curl/lib/libcurl.a exists. PASS
build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for dunfell_linux 
On the branch feature/52-dependant-libraries-fixed-versions. PASS
build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================
```

**Review by kanjoe24 — DISMISSED** [review]
approving

**Review by kanjoe24 — APPROVED** [review]
Approved.

**Comment by kanjoe24** [comment]
Test results:

```
 ./release-test-script-ut-control.sh -t feature/52-dependant-libraries-fixed-versions
==========================================================
RESULTS for ubuntu 
On the branch feature/52-dependant-libraries-fixed-versions. PASS
build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for VM-SYNC 
On the branch feature/52-dependant-libraries-fixed-versions. PASS
build/linux/curl/lib/libcurl.a exists. PASS
build/linux/openssl/lib/libssl.a exists. PASS 
host-tools/CMake-3.30.0/build/bin/cmake exists. PASS 
==========================================================
==========================================================
RESULTS for dunfell_arm 
On the branch feature/52-dependant-libraries-fixed-versions. PASS
build/arm/curl/lib/libcurl.a exists. PASS
build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for dunfell_linux 
On the branch feature/52-dependant-libraries-fixed-versions. PASS
build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================

```

**Pull request merged**